src/main/java/com/artipie/api/ManageUsers.java

```diff
@@ -56,123 +55,173 @@ public final class ManageUsers implements CrudUsers {
-                final String name = node.asScalar().value();
-                if (!uname.equals(name)) {
-                    all = all.add(name, names.yamlMapping(name));
-                }
-            }
-            final YamlMapping user = users.get().yamlMapping(uname);
-            YamlMappingBuilder changing = Yaml.createYamlMappingBuilder();
-            for (final YamlNode node : user.keys()) {
-                final String prop = node.asScalar().value();
-                if (!ManageUsers.TYPE.equals(prop) && !ManageUsers.PASS.equals(prop)) {
-                    changing = changing.add(prop, user.value(prop));
-                }
-            }
-            changing = changing.add(ManageUsers.PASS, info.getString("new_pass"));
-            changing = changing.add(ManageUsers.TYPE, info.getString("new_type"));
-            all = all.add(uname, changing.build());
+    /**
+     * Activate user.
+     * @param uname User to activate
+     * @param enable Enable or disable
+     */
+    private void activateUser(final String uname, final boolean enable) {
+        final Optional<YamlMapping> info = this.userInfo(uname);
+        if (info.isPresent()) {
+            YamlMappingBuilder builder = ManageUsers.copyUserInfo(info.get());
+            builder = builder.add("enabled", String.valueOf(enable));
             this.blsto.save(
-                this.key,
-                Yaml.createYamlMappingBuilder()
-                    .add(YamlSettings.NODE_CREDENTIALS, all.build())
-                    .build().toString().getBytes(StandardCharsets.UTF_8)
+                this.userFileKey(uname),
+                builder.build().toString().getBytes(StandardCharsets.UTF_8)
             );
-            return;
+        } else {
+            throw new IllegalStateException(String.format("User %s does not exists", uname));
         }
-        throw new IllegalStateException(String.format("User %s does not found", uname));
     }
 
     /**
-     * Read yaml mapping with users from yaml file.
-     *
-     * @return Users yaml mapping
+     * Read user info from yaml. Returns empty if user does not exist.
+     * @param uname The name of the user
+     * @return User info yaml
      */
-    private Optional<YamlMapping> users() {
-        Optional<YamlMapping> res = Optional.empty();
-        if (this.blsto.exists(this.key)) {
-            try {
-                res = Optional.ofNullable(
-                    Yaml.createYamlInput(
-                        new String(this.blsto.value(this.key), StandardCharsets.UTF_8)
-                    ).readYamlMapping().yamlMapping("credentials")
-                );
-            } catch (final IOException err) {
-                throw new UncheckedIOException(err);
-            }
+    private Optional<YamlMapping> userInfo(final String uname) {
+        Optional<byte[]> res = Optional.empty();
+        final Pair<Key, Key> keys = ManageUsers.keys(uname);
+        if (this.blsto.exists(keys.getLeft())) {
```

Comment:
Can we use: userFileKey ?

Reply:
no, because it does not check that the second key variant actually exists